dev-util/electron/electron-0.36.9.ebuild

```diff
@@ -0,0 +1,707 @@
+	sys-devel/flex
+	virtual/pkgconfig"
+
+# For nvidia-drivers blocker, see bug #413637 .
+RDEPEND+="
+	x11-misc/xdg-utils
+	virtual/opengl
+	virtual/ttf-fonts
+	selinux? ( sec-policy/selinux-chromium )
+	tcmalloc? ( !<x11-drivers/nvidia-drivers-331.20 )
+	widevine? ( www-plugins/chrome-binary-plugins[widevine(-)] )"
+
+# Python dependencies. The DEPEND part needs to be kept in sync
+# with python_check_deps.
+DEPEND+=" $(python_gen_any_dep '
+	dev-python/beautifulsoup:python-2[${PYTHON_USEDEP}]
+	dev-python/beautifulsoup:4[${PYTHON_USEDEP}]
+	dev-python/html5lib[${PYTHON_USEDEP}]
+	dev-python/jinja[${PYTHON_USEDEP}]
+	dev-python/ply[${PYTHON_USEDEP}]
+	dev-python/simplejson[${PYTHON_USEDEP}]
+')"
+python_check_deps() {
+	has_version --host-root "dev-python/beautifulsoup:python-2[${PYTHON_USEDEP}]" &&
+	has_version --host-root "dev-python/beautifulsoup:4[${PYTHON_USEDEP}]" &&
+	has_version --host-root "dev-python/html5lib[${PYTHON_USEDEP}]" &&
+	has_version --host-root "dev-python/jinja[${PYTHON_USEDEP}]" &&
+	has_version --host-root "dev-python/ply[${PYTHON_USEDEP}]" &&
+	has_version --host-root "dev-python/simplejson[${PYTHON_USEDEP}]"
+}
+
+if ! has chromium_pkg_die ${EBUILD_DEATH_HOOKS}; then
+	EBUILD_DEATH_HOOKS+=" chromium_pkg_die";
+fi
+
+pkg_pretend() {
+	if [[ $(tc-getCC)$ == *gcc* ]] && \
+		[[ $(gcc-major-version)$(gcc-minor-version) -lt 48 ]]; then
+		die 'At least gcc 4.8 is required, see bugs: #535730, #525374, #518668.'
+	fi
+
+	# LTO pass requires more file descriptors
+	if use lto; then
+		local lto_n_rlimit_min="16384"
+		local maxfiles=$(ulimit -n -H)
+		if [ "${maxfiles}" -lt "${lto_n_rlimit_min}" ]; then
+			eerror ""
+			eerror "Building with USE=\"lto\" requires file descriptor" \
+				   "limit to be no less than ${lto_n_rlimit_min}."
+		    eerror "The current limit for portage is ${maxfiles}."
+			eerror "Please add the following to /etc/security/limits.conf:"
+			eerror ""
+			eerror "   root hard    nofile  ${lto_n_rlimit_min}"
+			eerror "   root soft    nofile  ${lto_n_rlimit_min}"
+			eerror ""
+			die
+		fi
+	fi
+
+	# Check build requirements, bug #541816 and bug #471810 .
+	CHECKREQS_MEMORY="3G"
+	use lto && CHECKREQS_MEMORY="5G"
+	CHECKREQS_DISK_BUILD="10G"
+	eshopts_push -s extglob
+	if is-flagq '-g?(gdb)?([1-9])'; then
+		CHECKREQS_DISK_BUILD="25G"
+	fi
+	eshopts_pop
+	check-reqs_pkg_pretend
+}
+
+pkg_setup() {
+	# Make sure the build system will use the right python, bug #344367.
+	python-any-r1_pkg_setup
+
+	chromium_suid_sandbox_check_kernel_config
+}
+
+src_unpack() {
+	default_src_unpack || die
+
+	EGIT_COMMIT="v${PV}" \
+	EGIT_REPO_URI=${ELECTRON_REPO_URI} \
+		git-r3_src_unpack || die
+
+	# Fuse chromium and electron source together, as
+	# this seems to be the only reasonable way to build and configure
+	# everything in a single pass.
+	rsync -a "${WORKDIR}/${CHROMIUM_P}/" "${S}/" || die
+
+	cd "${S}" || die
+	einfo "Installing node modules required for Electron build..."
+	npm install || die
```

Comment:
> As the ebuild is currently written, we cannot introduce it in `~amd64`.
> Downloading source code in src_unpack is banned outside of "live" ebuilds.
> That ban also includes fetching sources from git.
> 
> Your best bet would be to create a tarball or tarball(s) containing all
> needed sources.

Done.  Turns out not all submodules are really needed.  I replaced the 
necessary ones with links to Github snapshots, hopefully that's OK.  I also 
refactored how sources are merged together to avoid excessive copying.